packages/webapp/components/invite/Referral.tsx

```diff
@@ -0,0 +1,221 @@
+    onSuccess: async () => {
+      await Promise.all([
+        completeAction(ActionType.AcceptedGenericReferral),
+        refetchBoot(),
+      ]);
+      router.push(redirectTo);
+    },
+    onError: (err: ApiErrorResult) => {
+      const message = err?.response?.errors?.[0]?.message;
+      displayToast(message ?? DEFAULT_ERROR);
+    },
+  });
+  const [isVideoOpen, setVideoOpen] = useState(false);
+
+  const onVideoModalClose = () => {
+    setVideoOpen(false);
+  };
+
+  const handleAcceptClick = () => {
+    const handleAccept = () =>
+      onAcceptMutation({
+        token,
+        referrerId: referringUser.id,
+        feature: ReferralCampaignKey.Generic,
+      });
+
+    if (!user) {
+      return showLogin({
+        trigger: AuthTriggers.GenericReferral,
+        options: {
+          onLoginSuccess: handleAccept,
+          onRegistrationSuccess: handleAccept,
+        },
+      });
+    }
+
+    // since in the page view, query params are tracked automatically,
+    // we don't need to send the params here explicitly
+    trackEvent({ event_name: AnalyticsEvent.AcceptGeneralReferralInvitation });
+
+    return handleAccept();
+  };
+
+  useEffect(() => {
+    if (genericReferral === GenericReferral.Control) {
+      return;
+    }
+
+    router.push(redirectTo);
+    // router is an unstable dependency
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [redirectTo, genericReferral]);
+
+  return (
+    <div
+      style={{
+        background: `url(${cloudinary.referralCampaign.genericReferral.backgroundDark})`,
+        backgroundSize: 'cover',
+      }}
+      className="flex overflow-hidden relative flex-col laptop:flex-row flex-auto items-center p-2 laptop:p-6 h-[100vh]"
+    >
+      <span className="absolute top-8 left-1/2 laptop:left-12 -translate-x-1/2 laptop:translate-x-0 scale-[1.375]">
+        <Logo showGreeting={false} position={LogoPosition.Relative} />
+      </span>
+      <div className="flex z-1 flex-col p-4 laptop:p-0 laptop:mt-0 laptop:ml-3 w-full laptopL:ml-[9.315rem] laptopL:max-w-[37.2rem] mt-[5rem]">
+        <span className="flex flex-col laptop:flex-row gap-3 laptopL:gap-6 items-center laptop:items-start mb-6 tablet:mb-8">
+          <ProfileImageLink
+            user={referringUser}
+            picture={{ size: isLaptopL ? 'xxxlarge' : 'xlarge' }}
+          />
+          <p className="my-auto text-center laptop:text-left typo-headline laptop:typo-title3 laptopL:typo-title2">
+            <span className="block laptop:inline">{referringUser.name}</span>
+            <span className="font-normal"> invited you to daily.dev</span>
+          </p>
+        </span>
+        <h1 className="w-full text-center laptop:text-left break-words-overflow typo-large-title tablet:typo-mega2 laptopL:typo-giga2">
+          The Homepage
+          <span className="block font-bold text-theme-color-cabbage">
+            {' '}
+            Developers Deserve
```

Comment:
Not your fault, but think the [capitalization](https://dailydotdev.slack.com/archives/C05P9ET7S9K/p1698997971107649) here is wrong.